WordPress/gutenberg
# Add defaultFontSizes option to theme.json

## What?


Adds a `defaultFontSizes` option to disable showing the default font sizes.

When `defaultFontSizes === false` the options are removed from the UI.

By default `defaultFontSizes` is `true`.

## Why?


Resolves #52200

## How?


- Added option in the places it was needed
- Updated the base `TypographyPanel` to remove defaults when the option was set to false

## Testing Instructions





Test the new `defaultFontSizes`.

1. Add `settings.typography.defaultFontSizes = false` to your theme.json file.
2. See that the default options do not show.
3. Remove the setting.
4. See that the default options do show again.

Test that duplicate font sizes are removed.

1. Update `settings.typography.fontSizes` to include duplicate slugs.
2. Ensure that only the first of the duplicates is included.

### Testing Instructions for Keyboard


N/A

## Screenshots or screencast 

Tested with the theme.json from the example in #52200 and `defaultFontSizes` set to `false`.

| Before | After |
| ------ | ----- |
| <img width="280" alt="image" src="https://github.com/WordPress/gutenberg/assets/5129775/33578516-6bf8-4055-909a-39f6919ea2d8"> | <img width="280" alt="image" src="https://github.com/WordPress/gutenberg/assets/5129775/7da46da1-e585-4b73-baa8-660ee6d7bb53"> |
